### DISCUSSION TITLE
Question: Can Uptime monitor internal hosts using Nginx Proxy on specific ports

### DISCUSSION THREAD

**Discussion by gripfastistech** [upptime/upptime#103] [Q&A] [answered]
First, Upptime is really cool. **Thank you**! Been looking for a straightforward tool like this for awhile...I'm really enjoying it so far.

I'm want to monitor a few machines that are not publicly hosted by using an Nginx reverse proxy on a specific port as a pass-through to internal systems.

These systems are servers, NAS devices, and hosts that often have their own internal dns/domain names by a local host controller/AD/DNS/static IP/etc and many already have open ports to the internet to provide data to the world.

I've also thought one could create a specific domain with a bunch of subdomains to monitor with the proxy, so if Upptime can ping the subdomain through the proxy Upptime believes its just another website and says "OK" its up and reports normally.

Any thoughts/advice/comments?

**Comment by AnandChowdhary** [comment, marked as answer]
In general, if you already have an Nginx proxy server that forwards to the specific site, it should work, right? For example if  http://example.com:3342 is a proxy for http://your-ip-address:port, then you can add the example.com URL as an Upptime site. Your strategy of adding subdomains also works, yes. Either way, as along as it's a URL open on the web, and it returns an HTTP code response, Upptime can work with it.

**Comment by gripfastistech** [comment]
@AnandChowdhary the first one worked, awesome, thanks!

**Comment by gripfastistech** [comment]
@AnandChowdhary Proxy site, shown here a DLC, doesn't seem to get a graph populated (?). See image.

![Screenshot_2020-12-03 Real-time and historical data on system performance ](https://user-images.githubusercontent.com/53624322/101106084-2a813c80-3584-11eb-8c58-00966b13dc48.png)

**Comment by AnandChowdhary** [comment]
I think the graph issue is due to #106, which I'm fixing. :)
